chia/full_node/mempool_manager.py

```diff
@@ -582,9 +595,13 @@ async def new_peak(
             )
             self.seen_bundle_hashes = {}
             for item in old_pool.spends.values():
-                _, result, err = await self.add_spend_bundle(item.spend_bundle, item.npc_result, item.spend_bundle_name)
+                _, result, err = await self.add_spend_bundle(
+                    item.spend_bundle, item.npc_result, item.spend_bundle_name, item.height_added_to_mempool
+                )
                 # Only add to `seen` if inclusion worked, so it can be resubmitted in case of a reorg
                 if result == MempoolInclusionStatus.SUCCESS:
+                    new_item = self.get_mempool_item(item.spend_bundle_name)
+                    assert item == new_item
```

Comment:
this looks like debugging left-overs. did you mean to leave it in?

Reply:
You know, it is debugging leftover, but I did intend to leave it in, as an invariant.

This invariant expresses the exact problem that this PR fixes - that the `MempoolItem` included into the "new" Mempool by `add_spendbundle` is different than the MempoolItem in the "old" Mempool when rebuilding the mempool during `new_peak`.

Reply:
are we calling get_mempool_item on the item we just added to see if they are the same ?
if thats the case i think i would do some kind of unittest instead of leaving it here

Reply:
Almog, you read my mind. I was just coming back to ask if you two thought I should move this out into a test.

Reply:
This is checked in `test_mempoolitem_height_added`